### PR TITLE
fix composer binary execution on macOS Sierra and composer installed …

### PR DESCRIPTION
**Pull request by JanneAalto**
…from homebrew

fixes https://github.com/heroku/heroku-buildpack-php/issues/179

### CODE REVIEW

**Comment by jussikinnula** [comment]
@dzuelke works for me, thanks @JanneAalto! 👍

**Comment by dzuelke** [comment]
Thanks for this. Turns out this doesn't just fix it for macOS!

Ubuntu 14.04:
> `a /usr/bin/env bash script, ASCII text executable, with very long lines`

Ubuntu 16.04:
> `Bourne-Again shell script, ASCII text executable, with very long lines`

And for the sake of documenting it:

Mac OS X 10.11:
> `a bash script text executable`

macOS 10.12:
> `POSIX shell script text executable`

**Comment by freeformz** [comment]
Why not just:

```console
if [ -x $composer_bin ]; then
  $composer_bin "$@"
else
  php -n $composer_bin "$@"
fi
```

**Comment by dzuelke** [comment]
Because `composer` could be (and commonly is) a PHAR archive (special `.php` package), and in that case, I want to run it through `php -n`, which prevents the loading of any config, so it will, on Heroku, also prevent New Relic from loading, sending bogus "CLI call" telemetry back, and outputting startup info into the logs.

File: bin/heroku-php-nginx [hunk 1]
```diff
@@ -201,7 +201,7 @@ unset -f php-fpm # remove the alias we made earlier that would prevent newrelic
 composer() {
     local composer_bin=$(which ./composer.phar composer | head -n1)
     # check if we the composer binary is executable by PHP
-    if file --brief --dereference $composer_bin | grep "bash" > /dev/null ; then # newer versions of file return "data" for .phar
+    if file --brief --dereference $composer_bin | grep "shell" > /dev/null || file --brief --dereference $composer_bin | grep "bash" > /dev/null ; then # newer versions of file return "data" for .phar
```
Comment (freeformz) [inline bin/heroku-php-nginx]: Try:
```console
if file --brief --dereference $composer_bin | grep -e "shell" -e "bash" > /dev/null; then # newer versions of file return "data" for .phar
```

**Comment by freeformz** [comment]
@dzuelke explained why not to me. But also, see comment, you can use multiple `-e pattern`'s to grep.

**Comment by dzuelke** [comment]
Please use `grep -e "bash" -e "shell"`, @JanneAalto, and make the same change in the other three boot scripts.

**Comment by JanneAalto** [comment]
@dzuelke changed.

**Comment by freeformz** [comment]
:+1: 

PS: Where are the tests?

**Pull request merged**